lib/widgets/type_of_leave_list.dart

```diff
@@ -0,0 +1,89 @@
+import 'package:cobe_hive_mobile_app/app_colors.dart';
+import 'package:cobe_hive_mobile_app/capitalize_string.dart';
+import 'package:cobe_hive_mobile_app/leave_request.dart';
+import 'package:cobe_hive_mobile_app/providers/leave_type_provider.dart';
+import 'package:flutter/material.dart';
+import 'package:hooks_riverpod/hooks_riverpod.dart';
+
+class TypeOfLeaveList extends ConsumerWidget {
+  const TypeOfLeaveList({super.key});
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    final selectedLeaveType = ref.watch(leaveTypeProvider);
```

Comment:
When a selected type leave changes, the whole list is rebuilt. This can be imrpoved by making another provider _(isLeaveSelectedProvider)_